app/cdap/components/Snackbar/index.tsx

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import Snackbar from '@material-ui/core/Snackbar';
+import Transition from 'components/Snackbar/Components/Transition/index';
+import { useStyles } from 'components/Snackbar/styles';
+import { ISnackbarProps } from 'components/Snackbar/types';
+import React, { useEffect, useState } from 'react';
+
+export default function({
+  handleCloseError,
+  description = '',
+  isSuccess,
+  snackbarAction,
+}: ISnackbarProps) {
+  const classes = useStyles();
+  const [isOpen, setIsOpen] = useState<boolean>(true);
```

Comment:
You are managing the state of this component twice - you have toaster.open used above... why not just pass that into this?

```
      {toaster.open && (
        <Snackbar
          handleCloseError={() =>
            setToaster({
              open: false,
            })
          }
          description={toaster.message}
          isSuccess={toaster.isSuccess}
        />
      )}{' '}
      ```
      
      Also the default value of isOpen is true but you're setting it to be true in the useEffect below so you're triggering a rerender. Either manage the state entirely outside of this component in the useSnackbar hook that i described or manage it in here as a reaction to props changing

Reply:
Also the default value of isOpen is true but you're setting it to be true in the useEffect below so you're triggering a rerender. Either manage the state entirely outside of this component in the useSnackbar hook that i described or manage it in here as a reaction to props changing

Reply:
Done